src/EditorFeatures/Core/Implementation/AutomaticCompletion/Sessions/EditorBraceCompletionSession.cs

```diff
@@ -0,0 +1,84 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Threading;
+using Microsoft.CodeAnalysis.BraceCompletion;
+using Microsoft.CodeAnalysis.Text;
+using Microsoft.VisualStudio.Text.BraceCompletion;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.Editor.Implementation.AutomaticCompletion.Sessions
+{
+    internal class EditorBraceCompletionSession : IEditorBraceCompletionSession
+    {
+        private readonly IBraceCompletionService _braceCompletionService;
+
+        public EditorBraceCompletionSession(IBraceCompletionService braceCompletionService)
+        {
+            _braceCompletionService = braceCompletionService;
+        }
+
+        public BraceCompletionResult? GetChangesAfterReturn(IBraceCompletionSession session, CancellationToken cancellationToken)
+        {
+            var context = GetBraceCompletionContextFromSession(session);
+            if (context == null)
+            {
+                return null;
+            }
+
+            return _braceCompletionService.GetTextChangeAfterReturnAsync(context, supportsVirtualSpace: true, cancellationToken).WaitAndGetResult(cancellationToken);
```

Comment:
in general, asynchrony should not be blocked at the leaves.  instead push the asynchrony up *as far as you can* and do the waiting explicitly at the highest level possible.  This keeps all the services/helpers/etc. nice and clean, and keeps the nasty unfortunate stuff isolated to one place.

Reply:
Since this type is basically now only used to call the async brace completion service from the sync BraceCompletionSession, I'll just move these calls up to BraceCompletionSession and delete this type